src/enrichment_facility_tests.cc

```diff
@@ -11,229 +11,484 @@
 }
 
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-TEST_F(EnrichmentFacilityTest, InitialState) {
-  EXPECT_EQ(in_recipe, src_facility->InRecipe());
-  EXPECT_EQ(in_commod, src_facility->in_commodity());
-  EXPECT_EQ(out_commod, src_facility->out_commodity());
-  EXPECT_DOUBLE_EQ(tails_assay, src_facility->TailsAssay());
-  EXPECT_DOUBLE_EQ(feed_assay, src_facility->FeedAssay());
-  EXPECT_DOUBLE_EQ(inv_size, src_facility->MaxInventorySize());
-  EXPECT_DOUBLE_EQ(0.0, src_facility->InventorySize());
-  EXPECT_DOUBLE_EQ(swu_capacity, src_facility->SwuCapacity());
-  EXPECT_EQ(reserves, src_facility->InitialReserves());
-}
-
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-TEST_F(EnrichmentFacilityTest, DISABLED_XMLInit) {
-  std::stringstream ss;
-  ss << "<start>"
-     << "<name>fooname</name>"
-     << "<config>"
-     << "<UNSPECIFIED>"
-     << "  <input>"
-     << "    <incommodity>" << in_commod << "</incommodity>"
-     << "    <inrecipe>" << in_recipe << "</inrecipe>"
-     << "    <inventorysize>" << inv_size << "</inventorysize>"
-     << "  </input>"
-     << "  <output>"
-     << "    <outcommodity>" << out_commod << "</outcommodity>"
-     << "    <tails_assay>" << tails_assay << "</tails_assay>"
-     << "    <swu_capacity>" << swu_capacity << "</swu_capacity>"
-     << "  </output>"
-     << "  <initial_condition>"
-     << "    <reserves_qty>" << reserves << "</reserves_qty>"
-     << "  </initial_condition>"
-     << "</UNSPECIFIED>"
-     << "</config>"
-     << "</start>";
-
-  cyclus::XMLParser p;
-  p.Init(ss);
-  cyclus::InfileTree engine(p);
-  cycamore::EnrichmentFacility fac(tc_.get());
-
-  // EXPECT_NO_THROW(fac.InitFrom(&engine););
-  EXPECT_EQ(in_recipe, fac.InRecipe());
-  EXPECT_EQ(in_commod, fac.in_commodity());
-  EXPECT_EQ(out_commod, fac.out_commodity());
-  EXPECT_DOUBLE_EQ(tails_assay, fac.TailsAssay());
-  EXPECT_DOUBLE_EQ(feed_assay, fac.FeedAssay());
-  EXPECT_DOUBLE_EQ(inv_size, fac.MaxInventorySize());
-  EXPECT_DOUBLE_EQ(0.0, fac.InventorySize());
-  EXPECT_DOUBLE_EQ(swu_capacity, fac.SwuCapacity());
-  EXPECT_EQ(reserves, fac.InitialReserves());
-}
-
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-TEST_F(EnrichmentFacilityTest, Clone) {
-  cyclus::Context* ctx = tc_.get();
-
-  cycamore::EnrichmentFacility* cloned_fac =
-    dynamic_cast<cycamore::EnrichmentFacility*>(src_facility->Clone());
-
-  EXPECT_EQ(in_recipe, cloned_fac->InRecipe());
-  EXPECT_EQ(in_commod, cloned_fac->in_commodity());
-  EXPECT_EQ(out_commod, cloned_fac->out_commodity());
-  EXPECT_DOUBLE_EQ(tails_assay, cloned_fac->TailsAssay());
-  EXPECT_DOUBLE_EQ(feed_assay, cloned_fac->FeedAssay());
-  EXPECT_DOUBLE_EQ(inv_size, cloned_fac->MaxInventorySize());
-  EXPECT_DOUBLE_EQ(0.0, cloned_fac->InventorySize());
-  EXPECT_DOUBLE_EQ(swu_capacity, cloned_fac->SwuCapacity());
-  EXPECT_EQ(reserves, cloned_fac->InitialReserves());
-
-  delete cloned_fac;
-}
-
```

Comment:
And since this is a code-generated cloning function,  we shouldn't be very concerned about testing it.